src/linpde_gp/randprocs/covfuncs/_linfunctls.py

```diff
@@ -118,15 +117,27 @@ def _(self, k: pn.randprocs.covfuncs.CovarianceFunction, /, *, argnum: int = 0):
 
 
 @LebesgueIntegral.__call__.register  # pylint: disable=no-member
-def _(self, k: Matern, /, *, argnum: int = 0):
+def _(self, k: pn.randprocs.covfuncs.CovarianceFunction, /, *, argnum: int = 0):
+    if argnum not in (0, 1):
+        raise ValueError("`argnum` must either be 0 or 1.")
+
+    from ..crosscov.linfunctls.integrals import (  # pylint: disable=import-outside-toplevel
+        CovarianceFunction_Identity_LebesgueIntegral,
+    )
+
+    return CovarianceFunction_Identity_LebesgueIntegral(k, self, reverse=(argnum == 0))
+
+
```

Comment:
```suggestion

@LebesgueIntegral.__call__.register # pylint: disable=no-member
def _(self, k: covfuncs.StackCovarianceFunction, /, *, argnum: int = 0):
    from ..crosscov import (  # pylint: disable=import-outside-toplevel
        StackedProcessVectorCrossCovariance,
    )

    return StackedProcessVectorCrossCovariance(
        *(self(covfunc, argnum=argnum) for covfunc in k.covfuncs)
    )


```

LebesgueIntegral(CovarianceFunction) has precedence over LinearFunctional(StackedCovarianceFunction). This is also the reason for the error in the notebook. We can fix this by explicitly reintroducing the rule for `LebesgueIntegral`.

Reply:
I think we can solve this problem a bit more generically. What do you think about the solution in #645c3a86?

Reply:
Beautiful :+1: